core/backend/src/test/standalone/RenderMaterialElement.test.ts

```diff
@@ -91,6 +91,52 @@ describe("RenderMaterialElement", () => {
       test({}, {});
     });
 
+    it("should convert TextureIds to hexadecimal string when loading element", () => {
```

Comment:
Please test the following cases:
- TextureId has a non-zero briefcase Id (e.g., 0x0000bc0000001234)
- TextureId has non-zero digits in the upper 32 bits (e.g., 0x000000123456789a)
- Combination of the two above
- TextureId is a numeric value outside the range of a `double` (e.g., 0xfeffff123456789a)
- The map has other properties besides `TextureId`, and those are preserved after native code converts `TextureId` from integer to Id string.

Reply:
Thanks, I believe I've added all of those test cases.

Reply:
Turns out my test is failing on [darwin arm64](https://dev.azure.com/bentleycs/iModelTechnologies/_build/results?buildId=3061563&view=logs&j=71a49582-1cc1-5c39-deb8-41612c740dcb&t=1d054115-f899-5a59-f14a-abc21087f657):
```
  1) RenderMaterialElement
       insert
         should be able to convert TextureId to hexadecimal string outside javascript's double precision range of 2^53:

      AssertionError: expected '0x7fffffffffffffff' to equal '0x8000000000000000'
  ...1 line omitted...

      -0x7fffffffffffffff
      +0x8000000000000000
      
      at Context.<anonymous> (src/test/standalone/RenderMaterialElement.test.ts:108:107)
      at processImmediate (node:internal/timers:476:21)
```

After looking into it with @tcobbs-bentley's help understanding the differences between intel/arm and float-to-int conversion...

#### Note for my test case:
I modified the jsonProperties of a database to have 2^63 represented as decimal for the textureId using a SQLite db browser for this test.

On the C++ side, I am using BeJsValue's GetId64 which use [GetUint64](https://github.com/iTwin/imodel-native/blob/nick/rendermaterial/iModelCore/Bentley/BeRapidJson/PublicAPI/BeRapidJson/BeJsValue.h#L274-L280).

This casts the number to a double then to an int64 then to a uint64. The initial cast to double and calling GetDouble already seems to be a problem if we are trying to represent numbers closer to that 2^63 limit, as that turns them into floating point numbers which as far as I understand can't represent those large numbers with very much precision.

**Question** for @jffmarker**, do we know for sure that connectors can put numbers in the 2^60 range with full precision into the jsonProperties? 

If the answer to that is yes, then we probably need to approach converting those numbers to hexadecimal strings slightly differently than what GetId64 does if possible, I.e. we can't convert to a double. Not sure if we can just call GetUInt64 on this and have things work? I see within BeRapidJsonUtilities we do have [Uint64FromValue](https://github.com/iTwin/imodel-native/blob/nick/rendermaterial/iModelCore/Bentley/BeRapidJson/PublicAPI/BeRapidJson/BeRapidJson.h#L35), but I don't know the full context behind some of the complexity in BeJsValue's GetId64.

#### Slightly less important but more context

Test passed on intel machines but failed on arm machines, due to the way each handles [float-to-int conversions](https://developer.apple.com/documentation/apple-silicon/addressing-architectural-differences-in-your-macos-code#Audit-Code-that-Contains-Float-to-Int-Conversions).
**For intel:**
- Converting 2^63 to a signed integer gives us the largest possible negative int which is -(2^63). 
- Converting back to unsigned gives us 2^63 
- Test passes because I expected to see 0x80000... ( which is 2^63).

**For arm:**
- Converting 2^63 to a signed integer gives us the largest possible positive int which is 2^63 - 1. 
- Converting back to unsigned gives us 2^63 -1.
- Test fails because we saw 0x7ffff.... (which is 2^63 -1)

Reply:
Just as a slight clarification to the architecture-specific float-to-int behavior.

When converting a double precision (64-bit) floating point value to a signed 64-bit integer on Intel Architecture, it behaves as expected for values that fit in the possible range of a 64-bit signed integer (-2^63 ... +2^63 - 1). However, if the floating point value is _larger_ than 2^63 - 1, it get converted to the minimum possible signed 64-bit integer (-2^63). That's right, all too-large _positive_ values end up being converted to the lowest possible (_negative_) value. I have no idea why this is the case. The Arm architecture instead converts the floating point value to the maximum possible _positive_ signed 64-bit integer (2^63-1). This is much more reasonable behavior, but since it is different than Intel, apparently it ends up causing bugs like this one.

Reply:
Yes, our C++ APIs (JsonCpp and RapidJson) can insert 64-bit signed or unsigned values with full precision. By default BeJsValue uses RapidJson, but often in the context of the addon it will use Napi::Value instead. DgnElement::m_jsonProps uses RapidJson. BeJsValue::operator=(uint64_t and int64_t) call pure virtual methods, but GetUint64 does not - it assumes we're dealing with JavaScript limits on precision (you can also find a comment on operator=(int64_t) claiming that the input must be < Number.MAX_SAFE_INTEGER, but nothing enforces this).

An option would be to add virtual functions to JsValueRef to retrieve 64-bit integer values. JsonCpp and RapidJson already support this. (JsonCpp asserts if the underlying value is a double, not an integer, and the double is larger than MAX_SAFE_INTEGER). NapiValueRef would have to implement these to convert from double to integer. It would return the default value if the double value is outside of the safe range for the requested type (signed/unsigned 64).

Additionally, we ought to consider deleting the bits of the API that permit storing 64-bit integers in JSON in both imodel-native and imodel02. Whether people are accidentally doing that for storing Ids, or just for storing plain integers, it always carries the risk of overflowing the constraints imposed by JavaScript.

Reply:
I believe I have implemented your suggestions on my [native PR](https://github.com/iTwin/imodel-native/pull/820/files).

I also created a backlog item reflecting your last suggestion to make it impossible to store 64-bit integers in JSON: https://github.com/iTwin/itwinjs-backlog/issues/1194